plugins/kapt3/kapt3-compiler/test/org/jetbrains/kotlin/kapt3/test/AbstractKotlinKapt3Test.kt

```diff
@@ -251,30 +252,25 @@ open class AbstractClassFileToSourceStubConverterTest : AbstractKotlinKapt3Test(
     fun testSuppressWarning() {}
 
     override fun doTest(filePath: String) {
-        val wholeFile = File(filePath)
+        val testFile = File(filePath)
 
-        kaptFlags.add(KaptFlag.MAP_DIAGNOSTIC_LOCATIONS)
+        kaptFlagsToAdd.add(KaptFlag.MAP_DIAGNOSTIC_LOCATIONS)
 
-        if (wholeFile.isOptionSet("CORRECT_ERROR_TYPES")) {
-            kaptFlags.add(KaptFlag.CORRECT_ERROR_TYPES)
-        }
-
-        if (wholeFile.isOptionSet("STRICT_MODE")) {
-            kaptFlags.add(KaptFlag.STRICT)
-        }
-
-        if (wholeFile.isOptionSet("STRIP_METADATA")) {
-            kaptFlags.add(KaptFlag.STRIP_METADATA)
+        addOrRemoveFlag(KaptFlag.CORRECT_ERROR_TYPES, testFile)
+        if (isFlagEnabled("STRICT_MODE", testFile)) {
+            kaptFlagsToAdd.add(KaptFlag.STRICT)
         }
+        addOrRemoveFlag(KaptFlag.STRIP_METADATA, testFile)
+        addOrRemoveFlag(KaptFlag.KEEP_KDOC_COMMENTS_IN_STUBS, testFile)
 
         super.doTest(filePath)
         doTestWithJdk9(AbstractClassFileToSourceStubConverterTest::class.java, filePath)
         doTestWithJdk11(AbstractClassFileToSourceStubConverterTest::class.java, filePath)
     }
 
     override fun check(kaptContext: KaptContextForStubGeneration, javaFiles: List<File>, txtFile: File, wholeFile: File) {
-        val generateNonExistentClass = wholeFile.isOptionSet("NON_EXISTENT_CLASS")
-        val validate = !wholeFile.isOptionSet("NO_VALIDATION")
+        val generateNonExistentClass = isFlagEnabled("NON_EXISTENT_CLASS", wholeFile)
+        val validate = !isFlagEnabled("NO_VALIDATION", wholeFile)
```

Comment:
It seems you have created isFlagDisabled for this case.

Reply:
Yes, but `isFlagDisabled` is not the same as`!isFlagEnabled`. `isFlagDisabled` checks the existence of the string `// !{flagName}` in a test file, whereas `!isFlagEnabled` checks the absence of the string `// {flagName}`. Here the existing code wants to use `!isFlagEnabled`, so I'm trying to keep that behavior.

Thank you for merging this PR!